### ISSUE TITLE
[Serve] Pydantic validation error for serve deployment

### ISSUE THREAD

**Issue by matthewfranglen** [ray-project/ray#17363]
### What is the problem?

The use of response_model in a FastAPI annotation on a serve.deployment annotated class causes a pydantic validation error.
This is due to the incorrect encoding of the model.

ray version 1.5.0.

### Reproduction (REQUIRED)

```python
import time
from typing import List

import ray
from ray import serve
from fastapi import FastAPI
from pydantic import BaseModel

class TestModel(BaseModel):
    a: str
    b: List[str]

ray.init(namespace="bug-repro")

app = FastAPI()

@serve.deployment(route_prefix="/")
@serve.ingress(app)
class TestDeployment:
    @app.get("/", response_model=TestModel)
    def test_endpoint(self):
        test_model = TestModel(a="a", b=["b"])
        return test_model

serve.start(detached=True, http_options={"port": 8005})
TestDeployment.deploy()

while True:
    time.sleep(10)
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

Request:

`curl localhost:8005/`

Logs:

```
2021-07-27 12:16:21,912 INFO services.py:1247 -- View the Ray dashboard at http://127.0.0.1:8265
(pid=2794977) 2021-07-27 12:16:23,027   INFO http_state.py:75 -- Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-node:192.168.1.54-0' on node 'node:192.168.1.54-0' listening on '127.0.0.1:8005'
2021-07-27 12:16:23,165 INFO api.py:716 -- Started detached Serve instance in namespace 'bug-repro'.
2021-07-27 12:16:23,171 INFO api.py:418 -- Updating deployment 'TestDeployment'.
(pid=2794976) INFO:     Started server process [2794976]
(pid=2794977) 2021-07-27 12:16:23,234   INFO backend_state.py:869 -- Adding 1 replicas to backend 'TestDeployment'.
(pid=2794976) ERROR:    Exception in ASGI application
(pid=2794976) Traceback (most recent call last):
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/uvicorn/protocols/http/h11_impl.py", line 369, in run_asgi
(pid=2794976)     result = await app(self.scope, self.receive, self.send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/uvicorn/middleware/proxy_headers.py", line 59, in __call__
(pid=2794976)     return await self.app(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/ray/serve/http_proxy.py", line 319, in __call__
(pid=2794976)     await self.starlette_router(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/routing.py", line 609, in __call__
(pid=2794976)     await self.default(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/ray/serve/http_proxy.py", line 303, in _fallback_to_prefix_router
(pid=2794976)     await _send_request_to_handle(handle, scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/ray/serve/http_proxy.py", line 60, in _send_request_to_handle
(pid=2794976)     result = await object_ref
(pid=2794976) ray.exceptions.RayTaskError(ValidationError): ray::RayServeWrappedReplica.handle_request() (pid=2794963, ip=192.168.1.54)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/ray/serve/backend_worker.py", line 122, in wrap_to_ray_error
(pid=2794976)     raise exception
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/ray/serve/backend_worker.py", line 242, in invoke_single
(pid=2794976)     result = await method_to_call(*args, **kwargs)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/ray/serve/api.py", line 1082, in __call__
(pid=2794976)     sender,
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/fastapi/applications.py", line 199, in __call__
(pid=2794976)     await super().__call__(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/applications.py", line 112, in __call__
(pid=2794976)     await self.middleware_stack(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/middleware/errors.py", line 181, in __call__
(pid=2794976)     raise exc from None
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/middleware/errors.py", line 159, in __call__
(pid=2794976)     await self.app(scope, receive, _send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
(pid=2794976)     raise exc from None
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/exceptions.py", line 71, in __call__
(pid=2794976)     await self.app(scope, receive, sender)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/routing.py", line 580, in __call__
(pid=2794976)     await route.handle(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/routing.py", line 241, in handle
(pid=2794976)     await self.app(scope, receive, send)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/starlette/routing.py", line 52, in app
(pid=2794976)     response = await func(request)
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/fastapi/routing.py", line 236, in app
(pid=2794976)     is_coroutine=is_coroutine,
(pid=2794976)   File "/home/matthew/.local/share/virtualenvs/entity-disambiguation-augmenter-EiR5HwE0/lib/python3.7/site-packages/fastapi/routing.py", line 130, in serialize_response
(pid=2794976)     raise ValidationError(errors, field.type_)
(pid=2794976) pydantic.error_wrappers.ValidationError: 1 validation error for TestModel
(pid=2794976) response -> b
(pid=2794976)   str type expected (type=type_error.str)
```

**Comment by matthewfranglen** [comment]
If the `response_model` argument is removed then the endpoint works perfectly, however the fast api documentation endpoints lack documentation regarding the response type.

**Comment by matthewfranglen** [comment]
The example code is just a slight modification of https://github.com/ray-project/ray/issues/16757 which may help with any fix?

**Comment by matthewfranglen** [comment]
It is related to having a List on the model, if I define the model as
```python
class TestModel(BaseModel):
    count: int
```

Then it works fine.

**Comment by simon-mo** [comment]
Nice find @matthewfranglen! I opened a fix here https://github.com/ray-project/ray/pull/17376 and it should be merged to master and it will automatically close this issue, one the issue closed you can try out the fix in ray nightly https://docs.ray.io/en/master/installation.html#daily-releases-nightlies and this fix will be part of 1.6.0 release (within 1month)

**Comment by matthewfranglen** [comment]
Wow that was fast! Thank you, I'm excited to try out the new release :)
